kernel/kernel-api/src/main/java/io/delta/kernel/expressions/predicates/Predicate.java

```diff
@@ -0,0 +1,172 @@
+ *    <li>Since version: 3.0.0</li>
+ *   </ul>
+ *  </li>
+ *  <li>Name: <code>&lt;&gt;</code>
+ *   <ul>
+ *    <li>SQL semantic: <code>expr1 &lt;&gt; expr2</code></li>
+ *    <li>Since version: 3.0.0</li>
+ *   </ul>
+ *  </li>
+ *  <li>Name: <code>&lt;</code>
+ *   <ul>
+ *    <li>SQL semantic: <code>expr1 &lt; expr2</code></li>
+ *    <li>Since version: 3.0.0</li>
+ *   </ul>
+ *  </li>
+ *  <li>Name: <code>&lt;=</code>
+ *   <ul>
+ *    <li>SQL semantic: <code>expr1 &lt;= expr2</code></li>
+ *    <li>Since version: 3.0.0</li>
+ *   </ul>
+ *  </li>
+ *  <li>Name: <code>&gt;</code>
+ *   <ul>
+ *    <li>SQL semantic: <code>expr1 &gt; expr2</code></li>
+ *    <li>Since version: 3.0.0</li>
+ *   </ul>
+ *  </li>
+ *  <li>Name: <code>&gt;=</code>
+ *   <ul>
+ *    <li>SQL semantic: <code>expr1 &gt;= expr2</code></li>
+ *    <li>Since version: 3.0.0</li>
+ *   </ul>
+ *  </li>
+ *  <li>Name: <code>ALWAYS_TRUE</code>
+ *   <ul>
+ *    <li>SQL semantic: <code>Constant expression whose value is `true`</code></li>
+ *    <li>Since version: 3.0.0</li>
+ *   </ul>
+ *  </li>
+ *  <li>Name: <code>ALWAYS_FALSE</code>
+ *   <ul>
+ *    <li>SQL semantic: <code>Constant expression whose value is `false`</code></li>
+ *    <li>Since version: 3.0.0</li>
+ *   </ul>
+ *  </li>
+ *  <li>Name: <code>AND</code>
+ *   <ul>
+ *    <li>SQL semantic: <code>expr1 AND expr2</code></li>
+ *    <li>Since version: 3.0.0</li>
+ *   </ul>
+ *  </li>
+ *  <li>Name: <code>OR</code>
+ *   <ul>
+ *    <li>SQL semantic: <code>expr1 OR expr2</code></li>
+ *    <li>Since version: 3.0.0</li>
+ *   </ul>
+ *  </li>
+ * </ol>
+ */
+public class Predicate extends ScalarExpression {
+    public Predicate(String name, List<Expression> children) {
+        super(name, children);
+        validatePredicateExpression(name, children);
+    }
+
+    @Override
+    public String toString() {
+        if (COMPARATOR_OPERATORS.contains(name)) {
+            return String.format("(%s %s %s)", children.get(0), name, children.get(1));
+        }
+        String args = children.stream().map(Object::toString)
+            .collect(Collectors.joining(","));
+        return String.format("%s(%s)", name, args);
+    }
+
+    private static final Set<String> COMPARATOR_OPERATORS =
+        Stream.of("=", ">", ">=", "<", "<=").collect(Collectors.toSet());
+    /**
+     * List of supported predicate expressions
+     */
+    private static final Set<String> PREDICATE_TYPES = Stream.of(
+        "=",
+        "<",
+        "<=",
+        ">",
+        ">=",
+        "AND",
+        "OR",
+        "ALWAYS_TRUE",
+        "ALWAYS_FALSE"
+    ).collect(Collectors.toSet());
+
+    /**
+     * Validate the predicate name is in the allowed list and expected children types.
+     *
+     * @param name Predicate expression name
+     * @param children List of input expressions to the predicate expression.
+     */
+    private static void validatePredicateExpression(String name, List<Expression> children) {
```

Comment:
@tdas I am wondering if we should get rid of this type checks? It is becoming complicated as there is a `ScalarExpression` in between `Predicate` and `Expression`.  `ScalarExpression` needs to aware of these types if it wants to validate the functions.

Reply:
ScalarExpression has no data type or type validation right? So whats the problem if Predicate constructor does type checks before calling in the ScalarExpression constructor which does not do type checks?

Reply:
So in future when we add more scalar expression that are not `Predicate`, where do we do the validation?

Reply:
The validation also includes the given `name` is one of the supported expression.

It looks like the validation is not adding much value. The implementation of the `ExpressionHandler` already throws error if it gets an expression which is not one of the listed expression in the JavaDoc of `Predicate` or `ScalarExpression`. I am inclined to remove these validation checks.

Reply:
If there is a case in the future where wrong expressions does not throw any error and silently does not produce the intended result.. then we throwing error might be better. Keep an eye out for those stuff.